addons/website_sale_stock/models/product_product.py

```diff
@@ -1,19 +1,50 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models
+from odoo import models, fields, _
 from odoo.http import request
 
 
 class ProductProduct(models.Model):
     _inherit = 'product.product'
 
+    stock_notification_partner_ids = fields.Many2many('res.partner', string='Back in stock Notifications')
```

Comment:
I would force a meaningful relation table name. There may have another m2m between products and partners.

Reply:
@KangOl You're right. Made the change :)